db/dataset.py

```diff
@@ -183,7 +183,15 @@ def get_public_datasets():
             row = dict(row)
             dataset_id = row["id"].encode('utf-8')
             row["status"] = dataset_status[dataset_id]
```

Comment:
If this dataset has not been submitted, it will not have a status. This will fail with a `KeyError`. You should use `dataset_status.get(dataset_id)` instead.